### PR TITLE
Upload checkpoints to platform

### PR DESCRIPTION
**Pull request by Louis-Dupont**

### CODE REVIEW

File: src/super_gradients/common/sg_loggers/deci_platform_sg_logger.py [hunk 1]
```diff
@@ -28,6 +28,7 @@ def __init__(
         training_params: dict,
         checkpoints_dir_path: str,
         model_name: str,
+        upload_model: bool = True,
```
Comment (ofrimasad) [inline src/super_gradients/common/sg_loggers/deci_platform_sg_logger.py:31]: not really related to this specific change. but can you please add parameters documentation

File: src/super_gradients/common/sg_loggers/deci_platform_sg_logger.py [hunk 2]
```diff
@@ -74,6 +75,18 @@ def upload(self):
         self._upload_latest_file_starting_with(start_with=CONSOLE_LOGS_PREFIX)
         self._upload_folder_files(folder_name=".hydra")
 
+        if self.upload_model:
```
Comment (ofrimasad) [inline src/super_gradients/common/sg_loggers/deci_platform_sg_logger.py:78]: This implementation will upload all pth files over and over again.
That is redundant and will take a lot of networks capacity.
Instead, override `add_checkpoint` function. This way, you will upload only new files.

**Review by ofrimasad — CHANGES_REQUESTED** [review]
override `add_checkpoint` instead of current implementation

File: src/super_gradients/common/sg_loggers/deci_platform_sg_logger.py (hunk 1 above)
Comment (Louis-Dupont) [reply src/super_gradients/common/sg_loggers/deci_platform_sg_logger.py:31]: I added doc for this logger, and at the same time updated other loggers which were missing some fields: https://github.com/Deci-AI/super-gradients/pull/799/commits/386678f10d0d2d85e0065ee4b7c095f8a86699be

File: src/super_gradients/common/sg_loggers/deci_platform_sg_logger.py (hunk 2 above)
Comment (Louis-Dupont) [reply src/super_gradients/common/sg_loggers/deci_platform_sg_logger.py]: Changed in https://github.com/Deci-AI/super-gradients/pull/799/commits/386678f10d0d2d85e0065ee4b7c095f8a86699be

**Review by ofrimasad — APPROVED** [review]
LGTM. very nice.

**Pull request merged**